code/ATMOSPHERICS/components/unary_devices/vent_scrubber.dm

```diff
@@ -181,21 +188,17 @@
 			var/datum/gas_mixture/filtered_out = new
 			filtered_out.temperature = removed.temperature
 			if(scrub_Toxins)
-				filtered_out.toxins = removed.toxins
-				removed.toxins = 0
+				filtered_out.gases[GAS_PL][MOLES] = removed.gases[GAS_PL][MOLES]
+				removed.gases[GAS_PL][MOLES] = 0
 			if(scrub_CO2)
-				filtered_out.carbon_dioxide = removed.carbon_dioxide
-				removed.carbon_dioxide = 0
-
-			if(removed.trace_gases.len>0)
-				for(var/datum/gas/trace_gas in removed.trace_gases)
-					if(istype(trace_gas, /datum/gas/oxygen_agent_b))
-						removed.trace_gases -= trace_gas
-						filtered_out.trace_gases += trace_gas
-					else if(istype(trace_gas, /datum/gas/sleeping_agent) && scrub_N2O)
-						removed.trace_gases -= trace_gas
-						filtered_out.trace_gases += trace_gas
-
+				filtered_out.gases[GAS_CO2][MOLES] = removed.gases[GAS_CO2][MOLES]
+				removed.gases[GAS_CO2][MOLES] = 0
+			if(TRUE) //no var for scrubbing agent b but I wanted the indentation to line up
```

Comment:
worst comes to worst I just remove the check lol; I just wanted it to be pretty :b